source/MaterialXRuntime/RtNodeDef.cpp

```diff
@@ -108,6 +108,18 @@ void RtNodeDef::setIsDefaultVersion(bool isDefault)
     v->getValue().asBool() = isDefault;
 }
 
+bool RtNodeDef::isVersionCompatible(const RtToken& version) const
+{
+    // Test if either the version matches or if no version passed in if this is the default version.
+    if (version == getVersion() ||
+        (version.str().empty() && getIsDefaultVersion()))
+    {
+        return true;
+    }
+    return false;
```

Comment:
Nitpick: could simplify to:

`return (version == getVersion() || (version.str().empty() && getIsDefaultVersion()));`